File: src/main/java/com/dh/ondot/member/api/response/LoginResponse.java
package com.dh.ondot.member.api.response;

public record LoginResponse(
        Long memberId,
        String accessToken,
        String refreshToken,
        boolean isNewMember
) {
    public static LoginResponse of(Long memberId, String accessToken, String refreshToken, boolean isNewMember) {
        return new LoginResponse(memberId, accessToken, refreshToken, isNewMember);
    }
}

File: src/main/kotlin/com/dh/ondot/member/presentation/response/LoginResponse.kt
package com.dh.ondot.member.presentation.response

import com.dh.ondot.member.application.dto.LoginResult

data class LoginResponse(
    val memberId: Long,
    val accessToken: String,
    val refreshToken: String,
    val isNewMember: Boolean,
) {
    companion object {
        fun from(result: LoginResult): LoginResponse =
            LoginResponse(result.memberId, result.accessToken, result.refreshToken, result.isNewMember)
    }
}
